### PR TITLE
Add C++ code for zigzag traversal of binary tree

### PR DESCRIPTION
**Pull request by Anishka0107**
**Changes:**

Added code for zigzag level order traversal (BFS) of a binary tree

### CODE REVIEW

File: code/data_structures/src/tree/binary_tree/binary_tree/traversal/zigzag/zigzag.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#include <bits/stdc++.h>
```
Comment (arnavb) [inline code/data_structures/src/tree/binary_tree/binary_tree/traversal/zigzag/zigzag.cpp]: NO NO NO; #1915

File: code/data_structures/src/tree/binary_tree/binary_tree/traversal/zigzag/zigzag.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+using namespace std;
+
+class Node {
+    public:
+        int info;
+        Node* lc;
+        Node* rc;
+        Node (int info) {
+            this->info = info;
+            lc = NULL;
```
Comment (arnavb) [inline code/data_structures/src/tree/binary_tree/binary_tree/traversal/zigzag/zigzag.cpp]: nullptr is better than NULL

File: code/data_structures/src/tree/binary_tree/binary_tree/traversal/zigzag/zigzag.cpp [hunk 3]
```diff
@@ -0,0 +1,61 @@
+#include <bits/stdc++.h>
+using namespace std;
```
Comment (arnavb) [inline code/data_structures/src/tree/binary_tree/binary_tree/traversal/zigzag/zigzag.cpp]: Recommend not using this.

File: code/data_structures/src/tree/binary_tree/binary_tree/traversal/zigzag/zigzag.cpp [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+int main() {
+    BT bt;
+    bt.root = new Node (1);
+    bt.root->lc = new Node (2);
+    bt.root->rc = new Node (3);
+    bt.root->lc->lc = new Node (4);
+    bt.root->lc->rc = new Node (5);
+    bt.root->rc->lc = new Node (6);
+    bt.root->rc->rc = new Node (7);
+    bt.zigzag_traversal();
```
Comment (arnavb) [inline code/data_structures/src/tree/binary_tree/binary_tree/traversal/zigzag/zigzag.cpp]: Your naming conventions are incorrect; Check the C++ Style Guide.

**Comment by Anishka0107** [comment]
I have addressed your comments in my next commit.

File: code/data_structures/src/tree/binary_tree/binary_tree/traversal/zigzag/zigzag.cpp [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+class Node
+{
+    public:
+        int info;
+        Node* left_child;
+        Node* right_child;
+        Node (int info)
+        {
+            this->info = info;
```
Comment (abdouskamel) [inline code/data_structures/src/tree/binary_tree/binary_tree/traversal/zigzag/zigzag.cpp]: Should use member initialization list here.

File: code/data_structures/src/tree/binary_tree/binary_tree/traversal/zigzag/zigzag.cpp [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        }
+};
+
+class BinaryTree
+{
+    public:
+        Node* root;
+        BinaryTree()
+        {
+            root = nullptr;
```
Comment (abdouskamel) [inline code/data_structures/src/tree/binary_tree/binary_tree/traversal/zigzag/zigzag.cpp]: Should use member initialization list also.

File: code/data_structures/src/tree/binary_tree/binary_tree/traversal/zigzag/zigzag.cpp (hunk 5 above)
Comment (arnavb) [reply code/data_structures/src/tree/binary_tree/binary_tree/traversal/zigzag/zigzag.cpp]: Actually, yes, you should use member initializer lists, as that is the general recommended practice.

File: code/data_structures/src/tree/binary_tree/binary_tree/traversal/zigzag/zigzag.cpp (hunk 6 above)
Comment (arnavb) [reply code/data_structures/src/tree/binary_tree/binary_tree/traversal/zigzag/zigzag.cpp]: Actually, yes, you should use member initializer lists, as that is the general recommended practice.

File: code/data_structures/src/tree/binary_tree/binary_tree/traversal/zigzag/zigzag.cpp (hunk 5 above)
Comment (abdouskamel) [reply code/data_structures/src/tree/binary_tree/binary_tree/traversal/zigzag/zigzag.cpp]: It's not present in C++ style guide, maybe we should add it ?

Comment (arnavb) [reply code/data_structures/src/tree/binary_tree/binary_tree/traversal/zigzag/zigzag.cpp]: True, you can consider making an addition to generally use initializer lists whenever possible. Good idea!

**Comment by Anishka0107** [comment]
I have addressed your comments in my next commit.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**